### PR TITLE
Fix export of redacted polls

### PR DESCRIPTION
**Pull request by rkouwenhoven**
This PR fixes vector-im/element-web#21337 by including the case when a poll is redacted for generating text. I used existing code from redacted messages in method ```textForMessageEvent```. Since this resulted in duplicate code, this was refactored into a function.

Signed-off-by: Robin Kouwenhoven <r.kouwenhoven@outlook.com>

---
This PR currently has no changelog labels, so will not be included in changelogs.

A reviewer can add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is, or add `Type: [enhancement/defect/task]` to the description and I'll add them for you.

Preview: https://pr8118--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/TextForEvent.tsx [hunk 1]
```diff
@@ -757,11 +748,38 @@ export function textForLocationEvent(event: MatrixEvent): () => string | null {
...
-    return () => _t("%(senderName)s has started a poll - %(pollQuestion)s", {
-        senderName: getSenderName(event),
-        pollQuestion: (event.unstableExtensibleEvent as PollStartEvent)?.question?.text,
-    });
+    return () => {
+        let message = '';
+
+        if (event.isRedacted()) {
+            message = textForRedactedPollAndMessageEvent(event);
+            const senderDisplayName = event.sender && event.sender.name ? event.sender.name : event.getSender();
```
Comment (kerryarchibald) [inline src/TextForEvent.tsx:772]: `event.sender?.name ?? event.getSender()`

**Review by kerryarchibald — COMMENTED** [review]
Can you add some test cases please?

File: src/TextForEvent.tsx (hunk 1 above)
Comment (rkouwenhoven) [reply src/TextForEvent.tsx]: I changed the line

**Comment by rkouwenhoven** [comment]
> Can you add some test cases please?

@kerryarchibald I added test cases for normal and redacted poll and message events

File: src/stores/widgets/ElementWidgetCapabilities.ts [hunk 2]
```diff
@@ -19,4 +19,9 @@ export enum ElementWidgetCapabilities {
      * @deprecated Use MSC2931 instead.
      */
     CanChangeViewedRoom = "io.element.view_room",
+    /**
```
Comment (kerryarchibald) [inline src/stores/widgets/ElementWidgetCapabilities.ts:22]: Don't think this belongs in this diff.

**Review by kerryarchibald — CHANGES_REQUESTED** [review]
Looks like your changes from https://github.com/matrix-org/matrix-react-sdk/pull/7935 are included here, please remove them.
Otherwise looks great, thanks for fixing this.

File: src/stores/widgets/ElementWidgetCapabilities.ts (hunk 2 above)
Comment (rkouwenhoven) [reply src/stores/widgets/ElementWidgetCapabilities.ts]: I reverted this commit

**Review by kerryarchibald — APPROVED** [review]
Thanks!

**Pull request merged**